### PR TITLE
Reload server on config changes

### PR DESCRIPTION
**Pull request by JacobMGEvans**
When making changes to the configuration file, the development server will restart. Currently `bindings` is passed into the server to be refreshed.

#439

### CODE REVIEW

**Comment by threepointone** [comment]
The checks were running for close to 3 hours so I cancelled them

**Comment by JacobMGEvans** [comment]
> The checks were running for close to 3 hours so I cancelled them

Yeah I tried restarting them a few times throughout the day... I don't know what is going on. They are running locally fine.

**Comment by threepointone** [comment]
Keep an eye on it the next time you do a push and stop it if it's going on for too long, or else we'll run out the free minutes we're allocated.

File: packages/wrangler/src/index.tsx [hunk 1]
```diff
@@ -1035,197 +1036,221 @@ function createCLIParser(argv: string[]) {
...
-          compatibilityDate={getDevCompatibilityDate(
-            config,
-            args["compatibility-date"]
-          )}
-          compatibilityFlags={
-            args["compatibility-flags"] || config.compatibility_flags
+        const getLocalPort = memoizeGetPort(DEFAULT_LOCAL_PORT);
+        const getInspectorPort = memoizeGetPort(9229);
+
+        const devComponent = async (configParams: Config) => {
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:1181]:
```suggestion
        async function getDevReactElement (configParam: Config) {
```

File: packages/wrangler/src/index.tsx [hunk 2]
```diff
@@ -1035,197 +1036,221 @@ function createCLIParser(argv: string[]) {
...
+          );
         }
-      }
 
-      // now log all available bindings into the terminal
-      printBindings({
-        ...bindings,
-        vars: maskedVars,
-      });
+        const getBindings = (configParam: Config) => ({
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:1131]:
```suggestion
        function getBindings (configParam: Config) { return {
```

File: packages/wrangler/src/index.tsx [hunk 3]
```diff
@@ -1035,197 +1036,221 @@ function createCLIParser(argv: string[]) {
...
+        let config = readConfig(configPath, args);
+
+        if (config.configPath) {
+          watcher = watch(config.configPath, {
+            persistent: true,
+          }).on("change", async (_event) => {
+            // TODO: Do we need to handle different `_event` types differently?
+            //       e.g. what if the file is deleted?
+            config = readConfig(config.configPath, args);
+            logger.log(`Wrangler.toml changed...`);
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:1055]:
```suggestion
            logger.log(`${path.basename(config.configPath)} changed...`);
```

**Review by threepointone — APPROVED** [review]
Looks good! dropped some nits, but approving anyway

File: packages/wrangler/src/index.tsx (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx:1131]: This change gives a linter error `Move function declaration to function body root.eslint[no-inner-declarations](https://eslint.org/docs/rules/no-inner-declarations)`

Comment (threepointone) [reply packages/wrangler/src/index.tsx:1131]: what happens if you move it up outside the `try` clause?

File: packages/wrangler/src/index.tsx [hunk 4]
```diff
@@ -1035,197 +1036,221 @@ function createCLIParser(argv: string[]) {
...
         }
-      }
 
-      // now log all available bindings into the terminal
-      printBindings({
-        ...bindings,
-        vars: maskedVars,
-      });
+        const getBindings = (configParam: Config) => ({
+          kv_namespaces: config.kv_namespaces?.map(
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:1132]:
```suggestion
          kv_namespaces: configParam.kv_namespaces?.map(
```
why does this not read from configParam?

File: packages/wrangler/src/index.tsx (hunk 2 above)
Comment (threepointone) [reply packages/wrangler/src/index.tsx:1131]: alternately, just disable that warning for this function

File: packages/wrangler/src/index.tsx [hunk 5]
```diff
@@ -1035,197 +1036,229 @@ function createCLIParser(argv: string[]) {
...
 
-      // now log all available bindings into the terminal
-      printBindings({
-        ...bindings,
-        vars: maskedVars,
-      });
+        // eslint-disable-next-line no-inner-declarations
+        async function getBindings(configParam: Config) {
+          return {
+            kv_namespaces: config.kv_namespaces?.map(
```
Comment (threepointone) [inline packages/wrangler/src/index.tsx:1136]: Should this config be configParam?

File: packages/wrangler/src/index.tsx (hunk 4 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx]: From my git diffs, it looks like it just got missed in the function conversion. Thanks for the catch!

File: packages/wrangler/src/index.tsx (hunk 5 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx:1136]: Yes. Just saw the previous comment updating now 😄

**Pull request merged**